src/analysis/processing/qgsrasteranalysisutils.cpp

```diff
@@ -285,5 +287,196 @@ void QgsRasterAnalysisUtils::applyRasterLogicOperator( const std::vector< QgsRas
   destinationRaster->setEditable( false );
 }
 
+std::vector<double> QgsRasterAnalysisUtils::getCellValuesFromBlockStack( const std::vector< std::unique_ptr< QgsRasterBlock > > &inputBlocks, int &row, int &col, bool &hasNoData )
+{
+  //get all values from inputBlocks
+  std::vector<double> cellValues;
+  cellValues.reserve( inputBlocks.size() );
+
+  for ( auto &block : inputBlocks )
+  {
+    double value = 0;
+    if ( !block || !block->isValid() )
+    {
+      hasNoData = true;
+      break;
+    }
+    else
+    {
+      value = block->valueAndNoData( row, col, hasNoData );
+      if ( hasNoData )
+        break;
+      cellValues.push_back( value );
+    }
+  }
+  return cellValues;
+}
+
+double QgsRasterAnalysisUtils::meanFromCellValues( std::vector<double> cellValues, int stackSize )
+{
+  double sum = std::accumulate( cellValues.begin(), cellValues.end(), 0.0 );
+  double mean = sum / static_cast<double>( stackSize );
+  return mean;
+}
+
+double QgsRasterAnalysisUtils::medianFromCellValues( std::vector<double> cellValues, int stackSize )
+{
+  std::sort( cellValues.begin(), cellValues.end() );
+  int medianElementIdx = stackSize / 2;
+  if ( stackSize % 2 == 0 )
+  {
+    return ( cellValues[medianElementIdx] + cellValues[medianElementIdx + 1] ) / 2.0;
+  }
+  else
+  {
+    return cellValues[medianElementIdx];
+  }
+}
+
+
+double QgsRasterAnalysisUtils::stddevFromCellValues( std::vector<double> cellValues, int stackSize )
+{
+  double mean = meanFromCellValues( cellValues, stackSize );
+  double accum = 0.0;
+  for ( int i = 0; i < stackSize; i++ )
+  {
+    accum += std::pow( ( cellValues.at( i ) - mean ), 2.0 );
+  }
```

Comment:
You could replace this with a call to varianceFromCellValues(...)

Reply:
Done